docs/_docs/reference/experimental/into.md

```diff
@@ -0,0 +1,139 @@
+
+```scala
+def concatAll(xss: into[IterableOnce[Char]]*): List[Char] =
+  xss.foldLeft(List[Char]())(_ ++ _)
+```
+Here, the call
+```scala
+concatAll(List('a'), "bc", Array('d', 'e'))
+```
+would apply two _different_ implicit conversions: the conversion from `String` to `Iterable[Char]` gets applied to the second argument and the conversion from `Array[Char]` to `Iterable[Char]` gets applied to the third argument.
+
+
+## Unwrapping `into`
+
+Since `into[T]` is an opaque type, its run-time representation is just `T`.
+At compile time, the type `into[T]` is a known supertype of the type `T`. So if `t: T`, then
+```scala
+  val x: into[T] = t
+```
+typechecks but
+```scala
+val y: T = x   // error
+```
+is ill-typed. We can recover the underlying type `T` using the `underlying` extension method which is also defined in object `Conversion`:
+```scala
+import Conversion.underlying
+
+val y: T = x.underlying    // ok
+```
+However, the next section shows that unwrapping with `.underlying` is not needed for parameters, which is the most common use case. So explicit unwrapping should be quite rare.
+
+
+
+## Dropping `into` for Parameters in Method Bodies
+
+The typical use cases for `into` wrappers are for parameters. Here, they specify that the
+corresponding arguments can be converted to the formal parameter types. On the other hand, inside a method, a parameter type can be assumed to be of the underlying type since the conversion already took place when the enclosing method was called. This is reflected in the type system which erases `into` wrappers in the local types of parameters
+as they are seen in a method body. Here is an example:
+```scala
+  def ++ (elems: into[IterableOnce[A]]): List[A] =
+    val buf = ListBuffer[A]()
+    for elem <- elems.iterator do // no `.underlying` needed here
+      buf += elems
+    buf.toList
+```
+Inside the `++` method, `elems` is of type `IterableOnce[A]`, not `into[IterableOne[A]]`. Hence, we can simply write `elems.iterator` to get at the `iterator` method of the `IterableOnce` class.
+
+Specifically, we erase all `into` wrappers in the local types of parameter types, on the top-level of these
+types as well as in all top-level co-variant subparts. Here, a part `S` of a type `T` is
+top-level covariant, if it is not enclosed in some type that appears in contra-variant or invariant position in `T`.
+
+## Into in Aliases
+
+Since `into` is a regular type constructor, it can be used anywhere, including in type aliases and type parameters. This gives a lot of flexibility to enable implicit conversions for user-visible types. For instance, the Laminar framework
+defined a type `Modifier` that is commonly used as a parameter type of user-defined methods and that should support implicit conversions into it. Pattern like this can be supported by defining a type alias such as
+```scala
+type Modifier = into[ModifierClass]
```

Comment:
allowing abstraction like this seems to lose the benefit of keeping it clear that a method argument might be implicitly converted - perhaps tooling™️ can fix this if scaladoc could track if argument types can be converted to

Reply:
Yes, maybe. But it was a concern raised by @sjrd, that it would be too cumbersome to require explicit `into` in user-defined Laminar functions. We can now address that concern. But we should do it only if there's a clear DSL-like pattern, IMO. Laminar modifiers might meet that requirement, but maybe not many other scenarios do.

Reply:
Personally i think this is fine, at least it communicates at the definition of the type that its a target